code/obj/submachine/weapon_vendor.dm

```diff
@@ -31,6 +31,7 @@
 	flags = TGUI_INTERACTIVE
 	object_flags = NO_GHOSTCRITTER //cry about it
 	layer = OBJ_LAYER - 0.1	// Match vending machines
+	var/track_credits = FALSE // True for nukies vendor, false for others
```

Comment:
This could just be an override of `ui_act` on the nukie vendor then

Reply:
hopefully this new change looks right?